plugins/performance-lab/includes/admin/rest-api.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * REST API integration for the plugin.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Namespace for performance-lab REST API.
+ *
+ * @var string
+ */
+const PERFLAB_REST_API_NAMESPACE = 'performance-lab/v1';
+
+/**
+ * Route for activating plugin.
+ *
+ * @var string
+ */
+const PERFLAB_ACTIVATE_PLUGIN_ROUTE = '/activate-plugin';
+
+/**
+ * Route for fetching plugin settings URL.
+ *
+ * @var string
+ */
+const PERFLAB_PLUGIN_SETTINGS_URL_ROUTE = '/plugin-settings-url';
+
+/**
+ * Registers endpoint for performance-lab REST API.
+ *
+ * @since n.e.x.t
+ * @access private
+ */
+function perflab_register_endpoint(): void {
+	register_rest_route(
+		PERFLAB_REST_API_NAMESPACE,
+		PERFLAB_ACTIVATE_PLUGIN_ROUTE,
+		array(
+			'methods'             => 'POST',
+			'args'                => array(
+				'slug' => array(
+					'type'        => 'string',
+					'description' => __( 'Plugin slug of plugin that needs to be activated.', 'performance-lab' ),
+					'required'    => true,
+				),
+			),
+			'callback'            => 'perflab_handle_activate_plugin',
+			'permission_callback' => static function () {
+				if ( current_user_can( 'install_plugins' ) ) {
+					return true;
+				}
+
+				return new WP_Error( 'cannot_install_plugin', __( 'Sorry, you are not allowed to install plugins on this site.', 'performance-lab' ) );
+			},
+		)
+	);
+
+	register_rest_route(
+		PERFLAB_REST_API_NAMESPACE,
+		PERFLAB_PLUGIN_SETTINGS_URL_ROUTE,
+		array(
+			'methods'             => 'POST',
```

Comment:
This endpoint is "idempotent" so it should use the GET method

```suggestion
			'methods'             => 'GET',
```